### ISSUE TITLE
choose a subreddit does not work on localhost

### ISSUE THREAD

**Issue by hashimshafiq** [Halnex/laravel-reddit#5]
After creating a subreddit, when I try to create a post it always says subreddit id fied required. It does not accept anything in that field.

**Comment by geosem42** [comment]
I haven't checked the app in a while but I just tested it online and locally and I did not get any errors while submitting new posts.
Can you please tell me of any errors in the console of your dev tools?

**Comment by hashimshafiq** [comment]
Thanks for your reply
Actually I am hosted this on xampp server localhost. When I submit post, it always says that 
**The subreddit id field is required.**
I created a subreddit and try to enter its name/id into that field but it is not accepting that and same error occurs again and again.
Then I go to your demo site then create a subreddit first and then create a post for it. When I enter subreddit name/id field then It shows the name of all the subreddis created and I can select it. And no error occurs and it post. But on lcalhost on my machine its not displaying the subreddits name/ids while creating a post. Why is that occuring. ?

**Comment by geosem42** [comment]
I honestly have no idea what could be causing this bug.

I tested it locally and online, it works without any errors.

Can you please give me the full error you are getting in the console area of your dev tools?

**Comment by hashimshafiq** [comment]
Thanks for replying.
http://i67.tinypic.com/33o6vlk.jpg
Here is the screenshot of what the error occurs.

**Comment by geosem42** [comment]
I have no idea why you'd get this error.

It is saying that it failed to load `jquery-1.10.2.min.js` but that's impossible because the app doesn't use such a file.

It uses `jquery.min.js` which is version `1.11.3`

I would suggest re-downloading the app and installing it again.

If that doesn't work, open up `create.blade.php` in `/resources/views/post/` and on lines 20 and 22 replace `data/subreddits` with your absolute URL `http://localhost/laravel/public/data/subreddits`

**Comment by hashimshafiq** [comment]
I have reinstalled it many times but got the same error.
Can yo please send me your app that you are using in localhost. 
`hashimshafiq0@gmail.com`

Thanks

**Comment by hashimshafiq** [comment]
Post problem is solved by putting absolute URL as you said @halnex but now another problem arises. Not more than 1 post is displayed or nor added in the database.

**Comment by DaGitNah** [comment]
Where you guys able to fix this? I tried deploying.. ran in to the same issue...

**Comment by hashimshafiq** [comment]
@DaGitNah which issue? If you are not able to select subreddits as I have this problem then this is the fix

 open up create.blade.php in /resources/views/post/ and on lines 20 and 22 replace data/subreddits with your absolute URL http://localhost/laravel/public/data/subreddits

**Comment by Michael-J-Scofield** [comment]
Shameless self promotion but since I didn't really like some stuff about this one (xss vulns, routing and other errors) I made my own from scratch.

Deployment should be as easy as deploying a normal Laravel project.

**repo:**
https://github.com/Michael-J-Scofield/plebbit

I'm still working on the project so feel free to check it out and leave suggestions or even open a pull requests.

**Comment by hashimshafiq** [comment]
Awesome. Looking forward for your project :)

**Comment by Michael-J-Scofield** [comment]
First version is already deployed. I'm working on some features like banning, alerts and private messaging :)

**Comment by geosem42** [comment]
@Michael-J-Scofield This is great stuff. Congrats on the project, it looks great.

Please keep in mind that this app is not meant to be deployed anywhere, certainly not in production. It is meant as an example of how things are done in Laravel.

An upgrade is a much needed thing and unfortunately, I do not have the time to work on it.

So I wish you the best luck on your version.

**Comment by Michael-J-Scofield** [comment]
Thanks 👍 your project inspired me to start my own so thank you for that ;)

**Comment by DaGitNah** [comment]
@Michael-J-Scofield  BRO! Did you get the sorting algorithm down? Looks like you can sort via hot, new and top! Thats awesome!

**Comment by Michael-J-Scofield** [comment]
@DaGitNah Yeah I did, when other things are done Ill probably tweak the algorithm for popular a bit but it works ;)

**Comment by DaGitNah** [comment]
@Michael-J-Scofield  does it have an admin account?

**Comment by DaGitNah** [comment]
Also, an advertising module would be nice... then it can be sustainable.

**Comment by Michael-J-Scofield** [comment]
@DaGitNah yes in the env file you can set the admin userid. 
What kind of advertising module do you mean. Since I'm not planning to run ads on this project.

**Comment by DaGitNah** [comment]
Sweet! Ok, for the advirtisement, its only two. One header image and other is another image on the side. users can bid to place ads for say amount of time...? Like an hour of 12 hours... higher hours = higher payments.

**Comment by Michael-J-Scofield** [comment]
No. If you want to run this with ads you're going to have to do it yourself since I'm not a fan of it. Would rather have upgrades, or a give gold thing.

**Comment by DaGitNah** [comment]
All about the gold thing then! :)
